src/nvim/option.c

```diff
@@ -3738,6 +3738,10 @@ did_set_string_option (
       if (varp == &p_enc && *curbuf->b_p_keymap != NUL)
         (void)keymap_init();
 
+      if (varp == &p_enc) {
+        ui_update_encoding();
```

Comment:
`&encoding` is a hint for Vim to recognise the text encoding of a file, and as such it is file specific; the help quoted above is saying that **changing the setting for a file will not change the file’s encoding itself** (i.e. it will not transcode the file). I’m a bit fuzzy as to how changes of that option affect this PR, but generally speaking, you cannot treat `&encoding` as a value to be set once on Vim startup.